src/components/Tables/DataTable/fragments/inputs/TableInputNumber.vue

```diff
@@ -0,0 +1,28 @@
+<template>
+  <InputNumber
+    :prefix="preset ? presets[preset].prefix : undefined"
+    :suffix="preset ? presets[preset].suffix : undefined"
+    :min-fraction-digits="
+      preset ? presets[preset].minFractionDigits : undefined
+    "
+    :format="!!preset"
+    :readonly="!edit"
+    :input-class="edit ? '' : '!border-none !shadow-none !bg-transparent'"
+  />
+</template>
+
+<script setup lang="ts">
+import InputNumber, { type InputNumberProps } from "primevue/inputnumber";
+
+export type TableInputNumberProps = {
+  edit: boolean;
+  preset?: null | "currency" | "percentage";
+};
+defineProps<TableInputNumberProps>();
+
+type presetsType = { [key: string]: InputNumberProps };
+const presets = <presetsType>{
```

Comment:
Dê preferência ao uso do tipo genérico (comentário geral da review). Além disso, corrija a case e tipe explicitamente:

```suggestion
type PresetsType = { [key: string]: InputNumberProps };
const presets: PresetsType = {
```

Além disso, aqui ocorre o mesmo que no `TableInputBoolean`, linha 16. Ao tiparmos corretamente essa key, tornamos o acesso a lista de presets mais seguro e removemos a necessidade de algumas modelagens (como a que ocorre nas linhas 3 e 6).